element/serialize.js

```diff
@@ -70,6 +84,51 @@ const SELF_CLOSING_TAGS = new Set( [
 	'wbr',
 ] );
 
+/**
+ * Inline element tags.
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements
+ *
+ * [ ...document.querySelectorAll( '.threecolumns code' ) ]
+ *     .map( ( el ) => el.textContent.replace( /(^<|>$)/g, '' ) )
+ *
+ * @type {Set}
+ */
+const INLINE_TAGS = new Set( [
+	'a',
+	'abbr',
+	'acronym',
+	'b',
+	'bdo',
+	'big',
+	'br',
+	'button',
+	'cite',
+	'code',
+	'dfn',
+	'em',
+	'i',
+	'img',
```

Comment:
It is strange to me that `img` is included in this list but not `audio`, `video`, `iframe` etc. Should it also be dropped here? Maybe we should limit this list to phrasing content excluding embedded content? This would also mean dropping form elements as it's not phrasing content. https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories

Reply:
>It is strange to me that `img` is included in this list but not `audio`, `video`, `iframe` etc. Should it also be dropped here?

As noted in [above comment](https://github.com/WordPress/gutenberg/pull/6716#discussion_r190337654), this list was generated from MDN's documentation and aligns precisely with the list described here:

https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements#Elements

More specifically, the specification describing the processing of whitespace makes a distinction on "inline formatting context": https://www.w3.org/TR/css-text-3/#white-space-rules

Trying to dig into where the basis for `img` as inline originated, I couldn't find the exact specification, and it may instead be on the basis of [`inline` being the initial value for the `display` CSS property](https://www.w3.org/TR/CSS21/visuren.html#display-prop) and most user-agent stylesheets ([[1]](https://github.com/sw4/revert.css), [[2]](https://drafts.csswg.org/css2/sample.html)) not otherwise specifying a `display` value for `img`. That said, these stylesheets also don't tend to include elements like `audio` and such,  and indeed in my testing browsers report their `display` as `inline`:

https://codepen.io/aduth/pen/WJBNBM?editors=1111

Maybe instead then we should invert this to be inclusive only on block content? Since this will be easier to maintain into the future, since new elements will be added whose default is not set and will assume the initial value of `inline`.

Reply:
Interesting: https://github.com/beautify-web/js-beautify/blob/56a6270bc0a66b0065841306e9ca137cf880874e/js/src/html/beautifier.js#L92-L102

Reply:
I'm inclined to follow the lead of the specification here, unless I've misinterpreted it, or there's some other reason js-beautify chose to use phrasing content (what's the relation between phrasing content and inline elements?).